trie/triedb/hashdb/database.go

```diff
@@ -621,8 +622,12 @@ func (db *Database) Scheme() string {
 }
 
 // Reader retrieves a node reader belonging to the given state root.
-func (db *Database) Reader(root common.Hash) *reader {
-	return &reader{db: db}
+// An error will be returned if the requested state is not available.
+func (db *Database) Reader(root common.Hash) (*reader, error) {
+	if _, err := db.Node(root); err != nil {
+		return nil, fmt.Errorf("state %x is not available", root)
```

Comment:
We are dropping `err` on the floor here, why not 
```suggestion
		return nil, fmt.Errorf("state %#x is not available: %v", root, err)
```
Also in general, I prefer to use `%#x` over `%x` , to get the `0x` prefix